pkg/crc/preflight/preflight_checks_tray_windows.go

```diff
@@ -19,130 +17,119 @@ import (
-	// wait for the script to finish executing
-	time.Sleep(2 * time.Second)
-	if err != nil {
-		logging.Debug("Failed to execute tray installation script")
-		return err
+	// Start the daemon process from the Posh script
+	if _, _, err := powershell.Execute(constants.DaemonPSscriptPath); err != nil {
+		return fmt.Errorf("Failed to start the daemon process: %w", err)
 	}
 
-	// check for 'success' file
-	if _, err = goos.Stat(filepath.Join(tempDir, "success")); goos.IsNotExist(err) {
-		return fmt.Errorf("Installation script didn't execute successfully: %v", err)
-	}
 	return nil
 }
 
-func escapeWindowsPassword(password string) string {
-	// escape specials characters (|`|$|"|') with '`' if present in password
-	if strings.Contains(password, "`") {
-		password = strings.ReplaceAll(password, "`", "``")
-	}
-	if strings.Contains(password, "$") {
-		password = strings.ReplaceAll(password, "$", "`$")
-	}
-	if strings.Contains(password, "\"") {
-		password = strings.ReplaceAll(password, "\"", "`\"")
+func fixTrayInstalled() error {
+	/* Start the tray process and copy the executable to start-up folder
+	 * "$USERPROFILE\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup"
+	 */
+	cmd := fmt.Sprintf(`New-Item -ItemType SymbolicLink -Path "%s" -Name "%s" -Value "%s"`,
+		constants.StartUpFolder,
+		constants.TrayShortcutName,
+		constants.TrayExecutablePath,
+	)
+	if _, _, err := powershell.ExecuteAsAdmin("Create symlink to tray in start-up folder", cmd); err != nil {
+		return fmt.Errorf("Error trying to create symlink to tray in start-up folder: %w", err)
 	}
-	if strings.Contains(password, "'") {
-		password = strings.ReplaceAll(password, "'", "`'")
+	cmd = fmt.Sprintf(`Start-Process -FilePath "%s"`, constants.TrayExecutablePath)
+	if _, _, err := powershell.Execute(cmd); err != nil {
+		return fmt.Errorf("Failed to start tray process: %w", err)
 	}
-	return password
+	return nil
 }
 
 func removeTray() error {
-	trayProcessName := constants.TrayExecutableName[:len(constants.TrayExecutableName)-4]
-
-	tempDir, err := ioutil.TempDir("", "crc")
-	if err != nil {
-		return err
+	var mErr errors.MultiError
+	if err := goos.Remove(filepath.Join(constants.StartUpFolder, constants.TrayShortcutName)); err != nil {
+		mErr.Collect(err)
+	}
+	if err := goos.Remove(constants.DaemonBatchFilePath); err != nil {
+		mErr.Collect(err)
+	}
+	if err := goos.Remove(constants.DaemonPSscriptPath); err != nil {
+		mErr.Collect(err)
 	}
-	defer func() {
-		_ = goos.RemoveAll(tempDir)
-	}()
 
-	psScriptContent := genTrayRemovalScript(
-		trayProcessName,
-		constants.TrayShortcutName,
-		constants.DaemonServiceName,
-		tempDir,
-	)
-	psFilePath := filepath.Join(tempDir, "trayRemoval.ps1")
+	logging.Info("Please run the following powershell command to finish the cleanup process.")
+	logging.Info("This will stop the CodeReady Containers daemon process: '(Get-Process crc).kill()'")
```

Comment:
Why aren't we doing it ourselves?

Reply:
Killing yourself? Hmmmm perhaps this was the issue?

Reply:
You should be able to detect the other process by the identifier. Though, this needs additional code.
Refinement or essential?

Reply:
So what is stopTray?

Reply:
@gbraad `stopTray` stops the tray process if one is running, it doesn't remove its symlink from the startup folder, in the `removeTray` we are removing those symlink from the startup folder

Reply:
> You should be able to detect the other process by the identifier. Though, this needs additional code.
> Refinement or essential?

Yes, was looking at this, will update with the additional code needed to handle this properly

Reply:
You are suggesting:

```
(Get-Process crc).kill()
```
and yu have a `stopTray` function... so this puzzles me ;-)

Reply:
the tray process is different from the daemon, `stopTray`closes the tray but not the crc daemon, that command was to kill the daemon process, but in the latest commit, this is removed, and the daemon is killed during cleanup by finding its pid